make.py

```diff
@@ -193,29 +193,32 @@
 
 print "=== Copying apps ==="
 for app in CFE_APPS:
-    dest = COMPOSITE_IMPL_NO_INTERFACE_DIR + app + "/cFE_"
-    sp.check_call("cp build/cpu1/osk_app_fw/*.o " + COMPOSITE_IMPL_NO_INTERFACE_DIR + app + "/",
+    try:
+        dest = COMPOSITE_IMPL_NO_INTERFACE_DIR + app + "/cFE_"
+        sp.check_call("cp build/cpu1/osk_app_fw/*.o " + COMPOSITE_IMPL_NO_INTERFACE_DIR + app + "/",
                   shell=True, cwd=CFE_DIR)
-    # Copying cfs lib object to each app.
-    shutil.copy('build/cpu1/cfs_lib/cfs_utils.o', dest + 'cfs_utils.o')
-    # Copying expat lib objects to each app.
-    shutil.copy('build/cpu1/expat_lib/expat_init.o', dest + 'expat_init.o')
-    shutil.copy('build/cpu1/expat_lib/xmlparse.o', dest + 'xmlparse.o')
-    shutil.copy('build/cpu1/expat_lib/xmlrole.o', dest + 'xmlrole.o')
-    shutil.copy('build/cpu1/expat_lib/xmltok.o', dest + 'xmltok.o')
-    shutil.copy('build/cpu1/expat_lib/xmltok_impl.o', dest + 'xmltok_impl.o')
-    shutil.copy('build/cpu1/expat_lib/xmltok_ns.o', dest + 'xmltok_ns.o')
-    for src in glob.glob(CFE_MAKE_ROOT + app + "/*.o"):
-        basename = os.path.basename(src)
-        print "Copying object '{}' to '{}'.".format(basename, dest + basename)
-        shutil.copy(src, dest + basename)
+        # Copying cfs lib object to each app.
+        shutil.copy('build/cpu1/cfs_lib/cfs_utils.o', dest + 'cfs_utils.o')
+        # Copying expat lib objects to each app.
+        shutil.copy('build/cpu1/expat_lib/expat_init.o', dest + 'expat_init.o')
+        shutil.copy('build/cpu1/expat_lib/xmlparse.o', dest + 'xmlparse.o')
+        shutil.copy('build/cpu1/expat_lib/xmlrole.o', dest + 'xmlrole.o')
+        shutil.copy('build/cpu1/expat_lib/xmltok.o', dest + 'xmltok.o')
+        shutil.copy('build/cpu1/expat_lib/xmltok_impl.o', dest + 'xmltok_impl.o')
+        shutil.copy('build/cpu1/expat_lib/xmltok_ns.o', dest + 'xmltok_ns.o')
+        for src in glob.glob(CFE_MAKE_ROOT + app + "/*.o"):
+            basename = os.path.basename(src)
+            print "Copying object '{}' to '{}'.".format(basename, dest + basename)
+            shutil.copy(src, dest + basename)
+    except:
+        pass # ignore err and keep going
```

Comment:
I think swallowing errors without reporting them is generally bad practice. It might be best just to do something like:
```
if !exists(dest):
    print("skipping {}, as it does not exist in your composite version".format(app))
    continue
```
at the top of the loop.  (This is obviously pseudocode.)

I think this is more readable/explicit, and will give the user more of an idea of what's going on

Reply:
Seems reasonable. However, doing IO operation without any try catch is bad practice as well since there are trillions of ways that an IO operation could fail.

Reply:
I guess I will still have the things wrapped by a try/catch, and add dest checking at the top.

Reply:
I think if an IO operation fails for some strange reason, the script should fail fast and propagate that to the user. That's another reason to avoid using the `try/catch` solution

Reply:
Although that's true, but `make` for composite won't stop even if this script fail fast. Leaving a print statement propagate a (or more) app copying has failed maybe a better chance.

Reply:
I think maybe best steps are:
- add an if statement for explicit skipping (and print a message)
- add try catch for IOErrors, making sure to print out the caught error
Does that sound good?

Reply:
Yep, sounds good. Doing it right now. I will rebase the commits along with other changes.

Reply:
How's now?